pkg/apis/config/defaults.go

```diff
@@ -68,65 +68,185 @@ func NewDefaultsConfigFromConfigMap(config *corev1.ConfigMap) (*Defaults, error)
+		}
 	}
-	return nil, errors.New("Defaults for Broker Configurations have not been set up.")
+
+	// Check cluster default configuration
+	return d.getClusterDefaultBrokerConfig(brokerClassName)
+}
+
+// getBrokerConfigForEmptyClassName returns the BrokerConfig for the given namespace when brokerClassName is empty.
+// It first checks the namespace specific configuration, then the cluster default configuration.
+func (d *Defaults) getBrokerConfigForEmptyClassName(ns string) (*BrokerConfig, error) {
+	// Check if namespace has a default broker class
+	if nsConfig, ok := d.NamespaceDefaultsConfig[ns]; ok && nsConfig != nil {
+		if nsConfig.DefaultBrokerClass != "" {
+			return d.getBrokerConfigByClassName(ns, nsConfig.DefaultBrokerClass)
+		}
+	}
+
+	// Fallback to cluster default configuration
+	return d.getClusterDefaultBrokerConfig("")
+}
+
+// getClusterDefaultBrokerConfig returns the BrokerConfig for the given brokerClassName.
+func (d *Defaults) getClusterDefaultBrokerConfig(brokerClassName string) (*BrokerConfig, error) {
+	if d.ClusterDefaultConfig == nil || d.ClusterDefaultConfig.BrokerConfig == nil {
+		return nil, errors.New("Defaults for Broker Configurations have not been set up.")
+	}
+
+	if brokerClassName == "" {
+		return d.ClusterDefaultConfig.BrokerConfig, nil
+	}
+
+	// Check if the brokerClassName is the default broker class for the whole cluster
+	if d.ClusterDefaultConfig.DefaultBrokerClass == brokerClassName {
+		return d.ClusterDefaultConfig.BrokerConfig, nil
+	}
+
+	if config, ok := d.ClusterDefaultConfig.BrokerClasses[brokerClassName]; ok && config != nil {
+		return config, nil
+	}
+
+	return d.ClusterDefaultConfig.BrokerConfig, nil
 }
 
 // GetBrokerClass returns a namespace specific Broker Class, and if
-// that doesn't exist, return a Cluster Default and if that doesn't exist
+// that doesn't exist, return a Cluster Default and if the defaults doesn't exist
 // return an error.
 func (d *Defaults) GetBrokerClass(ns string) (string, error) {
 	if d == nil {
 		return "", errors.New("Defaults are nil")
 	}
-	value, present := d.NamespaceDefaultsConfig[ns]
-	if present && value.BrokerClass != "" {
-		return value.BrokerClass, nil
+
+	// Check if the namespace has a specific configuration
+	if nsConfig, ok := d.NamespaceDefaultsConfig[ns]; ok && nsConfig != nil {
+		if nsConfig.DefaultBrokerClass != "" {
+			return nsConfig.DefaultBrokerClass, nil
+		}
 	}
-	if d.ClusterDefault != nil && d.ClusterDefault.BrokerClass != "" {
-		return d.ClusterDefault.BrokerClass, nil
+
+	// Fallback to cluster default configuration if namespace specific configuration is not set
+	if d.ClusterDefaultConfig != nil && d.ClusterDefaultConfig.DefaultBrokerClass != "" {
+		return d.ClusterDefaultConfig.DefaultBrokerClass, nil
 	}
-	return "", errors.New("Defaults for Broker Configurations have not been set up.")
+
+	// If neither namespace specific nor cluster default broker class is found
+	return "", fmt.Errorf("Defaults are nil")
```

Comment:
Can we make this error message more descriptive? Just saying `Defaults are nil` doesn't feel like it would be very helpful for debugging